z3c/dependencychecker/db.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class ImportsDatabase(object):
+
+    def __init__(self):
+        self._requirements = []
+        self._extras_requirements = {}
+
+    def set_requirements(self, requirements_list):
+        for requirement in requirements_list:
+            if self._requirement_already_exists(requirement):
+                logger.warn(
+                    '"%s" is declared twice on setup.py',
+                    requirement.name,
+                )
+            else:
+                self._requirements.append(requirement)
+
+    def _requirement_already_exists(self, requirement):
+        names = self.get_requirements_names()
+        return bool(requirement.name in names)
+
+    def get_requirements_names(self):
+        return [
+            x.name
+            for x in self._requirements
+        ]
+
+    def set_extra_requirements(self, extra_name, dotted_names):
+        """A bit of extra work needs to be done as pkg_resources API returns
+        all common requirements as well as the extras when asking for an extra
+        requirements.
+        """
+        only_extra_dotted_names = self._remove_duplicates(dotted_names)
+        if extra_name in self.get_extras():
+            new_dotted_names = self._only_new_requirements(
+                extra_name,
+                dotted_names,
+            )
+            self._extras_requirements[extra_name] += new_dotted_names
+
+            logger.warn(
+                'extra requirement "%s" is declared twice on setup.py',
+                extra_name,
+            )
+        else:
+            self._extras_requirements[extra_name] = only_extra_dotted_names
+
+    def _remove_duplicates(self, imports, other=None):
+        other_names = other
+        if other is None:
+            other_names = self.get_requirements_names()
+
+        filtered_list = [
+            x
```

Comment:
Is it clearer to use `requirement` instead of `x`? You're calling `.name` on it and that's something you can call on requirements, right?
With a proper name I wouldn't have to guess :-)

Reply:
As I noticed that everything is a DottedName, there is no need to use ``.name`` on them, the dunder methods can already take care of it.

But you are right, I renamed x for ``dotted_name``